tests/test_obslog_command.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use common::TestEnvironment;
+use regex::Regex;
+
+pub mod common;
+
+#[test]
+fn test_obslog_with_or_without_diff() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new", "-m", "my description"]);
+    std::fs::write(repo_path.join("file1"), "foo\nbar\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["rebase", "-r", "@", "-d", "root"]);
+
+    let stdout = get_log_output(&test_env, &repo_path, &["obslog"]);
+    insta::assert_snapshot!(stdout, @r###"
+    @ 813918f7b4e6 test.user@example.com 2001-02-03 04:05:08.000 +07:00    conflict
```

Comment:
How about a case where the conflict was resolved? Do we want to test showing the diff of how it was resolved?